### DISCUSSION TITLE
Bitdefender Script

### DISCUSSION THREAD

**Discussion by NotYoAverageJoe** [amidaware/tacticalrmm#1837] [General]
Hello, the Bitdefender script used to work pre recent update. Currently, when the script is deployed it starts on remote computer but sits idle four hours.

Ususally the install would run immediately. Tested on two different TRRM deployments and different computers.
![image](https://github.com/amidaware/tacticalrmm/assets/64831521/efbd3a3f-34ee-4734-803a-13199d917cd2)

**Comment by dinger1986** [comment]
these are community provided scripts looks like originally provided by @silversword411 and some updates by @NiceGuyIT possibly they have some input. But its possible that someone who uses bitdefender needs to look at it or provide access to it so it can be fixed

**Reply by NotYoAverageJoe** [reply]
Maybe they can comment here with a possible solution.

**Reply by silversword411** [reply]
I just posted what was shared in Discord.

I don't use 3rd party av and advise against wasting money on it when built in and free works just fine.

**Reply by NotYoAverageJoe** [reply]
Built in as in Defender?

**Reply by silversword411** [reply]
Yes

**Reply by NiceGuyIT** [reply]
The install will hang if there's other AV (not Defender) installed because it's prompting something about the other AV.

If you have only Defender and it's hanging, run it from an elevated PowerShell and see if it prompts for anything.

**Reply by NotYoAverageJoe** [reply]
I'm not good with PS. I modified the script and added the URL as reference.

It installed from PS no issues.

**Reply by NiceGuyIT** [reply]
Bitdefender should create a log of the install. You may be able to find out why it didn't work when run from TRMM. I believe the log is in TEMP.

**Reply by NotYoAverageJoe** [reply]
Thank you for directing me, the only log file I could locate in the temp dir related to Bitdefender install is this:

![image](https://github.com/amidaware/tacticalrmm/assets/64831521/95db16eb-2f86-426f-981c-a8e77839fccb)

It works when running it as the logged on user. The logged in user is admin, will have to test on standard account.

![image](https://github.com/amidaware/tacticalrmm/assets/64831521/23d16f3c-8f7d-4a44-8210-bf697cde12d5)

**Reply by NiceGuyIT** [reply]
The `-exe {{client.bdexe}}` parameter is not necessary because the script extracts the EXE from the URL. Uncheck Run As User because the installer requires Administrator permissions.

If it still hangs/fails, check `C:\Windows\Temp` for a directory "bd" directory or similar **while it's still running**. That contains the extracted version that it's trying to install. I believe it also contains a log file somewhere in there.

**Reply by NotYoAverageJoe** [reply]
I removed -exe {{client.bdexe}} and tried again.

Cleaned temp directory. All I see is this when the installation is running, in those folders I dont have any log file except the one share previously.

![image](https://github.com/amidaware/tacticalrmm/assets/64831521/0c8870c1-6eb9-48ba-89c1-067deeb51179)

The log file in this directory is from MS Office.
